store/tikv/region_cache.go

```diff
@@ -80,35 +86,21 @@ func (c *RPCContext) GetStoreID() uint64 {
 	return 0
 }
 
-func (c *CachedRegion) isValid() bool {
-	lastAccess := atomic.LoadInt64(&c.lastAccess)
-	lastAccessTime := time.Unix(lastAccess, 0)
-	return time.Since(lastAccessTime) < rcDefaultRegionCacheTTL
-}
-
-// GetCachedRegion returns a valid region
-func (c *RegionCache) GetCachedRegion(id RegionVerID) *Region {
-	c.mu.RLock()
-	cachedregion, ok := c.mu.regions[id]
-	c.mu.RUnlock()
-	if !ok {
-		return nil
-	}
-	if cachedregion.isValid() {
-		atomic.StoreInt64(&cachedregion.lastAccess, time.Now().Unix())
-		return cachedregion.region
-	}
-	c.DropRegion(id)
-	return nil
-}
-
 // GetRPCContext returns RPCContext for a region. If it returns nil, the region
 // must be out of date and already dropped from cache.
 func (c *RegionCache) GetRPCContext(bo *Backoffer, id RegionVerID) (*RPCContext, error) {
-	region := c.GetCachedRegion(id)
+	c.mu.RLock()
+	region := c.getCachedRegion(id)
 	if region == nil {
+		c.mu.RUnlock()
 		return nil, nil
 	}
+	// Note: it is safe to use region.meta and region.peer without clone after
+	// unlock, because region cache will never update the content of region's meta
+	// or peer. On the contrary, if we want to use `region` after unlock, then we
+	// need to clone it to avoid data race.
+	meta, peer := region.meta, region.peer
```

Comment:
If the `region` never modifies its content, access them with a `region` pointer is safe too.

Reply:
`region.peer` may be changed to another peer later, but the content of `reigon.peer` will not be changed.